seahorn/lib/nd_klee.c

```diff
@@ -45,3 +117,22 @@ void __VERIFIER_error(void) {
   klee_assert(0);
 }
 
+/* memhavoc - Special function for memory block initialization. 
+  * initialize memory content as non-deterministic value.
+*/
+void memhavoc(void *ptr, size_t size) {
+  for (size_t i = 0; i < size; ++i){
+    void *tmp_ptr = (ptr + i);
+    klee_make_symbolic(&tmp_ptr, sizeof(tmp_ptr), "havoc_content");
```

Comment:
this call to `klee_make_symbolic` is wrong in many levels.
At each iteration you write 4 or 8 bytes (depending on the architecture).